custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023-2024 Broadcom. All rights reserved.
+
+package dev.gemfire.stat.logger;
+
+import dev.gemfire.stat.exception.GemfireStatNotFoundException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.geode.*;
+import org.apache.geode.cache.*;
+import org.apache.geode.cache.client.internal.Op;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionContext;
+import org.apache.geode.distributed.DistributedSystem;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+
+public class StatLogger implements Function {
+
+  private static final String DEFAULT_LOG_FREQUENCY = "5000";
+  private static final String REGION = "GemFireStatLoggerTimer";
+  private static final String STATS_LOGGER_TIMER_REGION_KEY = "statsLoggerTimer";
+  private static final String STAT_LOGGER_THREAD_NAME = "StatLogger";
+  private static final String LOG_FREQUENCY_PROPERTY_NAME = "logFrequency";
+  private static final String STATISTICS_PROPERTY_NAME = "statistics";
+  private static final Logger LOGGER = LogService.getLogger();
+
+
+  @Override
+  public boolean isHA() {
+    return false;
+  }
+
+  @Override
+  public boolean hasResult() {
+    return true;
+  }
+
+  @Override
+  public String getId() {
+    return StatLogger.class.getSimpleName();
+  }
+
+  @Override
+  public void execute(FunctionContext functionContext) {
+    Properties properties = getProperties();
+    String logFrequency = properties.getProperty(LOG_FREQUENCY_PROPERTY_NAME, DEFAULT_LOG_FREQUENCY);
+    String statistics = properties.getProperty(STATISTICS_PROPERTY_NAME);
+
+    if (StringUtils.isEmpty(statistics)) {
+      cancelTimer(functionContext.getCache());
+      functionContext.getResultSender().lastResult("No properties set for StatLogger. StatLogger will not start.");
+      return;
+    }
+
+    long timerInterval = Long.parseLong(logFrequency);
```

Comment:
It's possible that this `parseLong()` call could throw an exception if the log frequency is not a valid long, so there should be some error handling here.

Reply:
Done!